misc/python/materialize/data_ingest/data_type.py

```diff
@@ -0,0 +1,187 @@
+        else:
+            raise ValueError(f"Unexpected record size {record_size}")
+
+    @staticmethod
+    def num_value(num: int) -> Any:
+        return num
+
+    @staticmethod
+    def name(backend: Backend) -> str:
+        if backend == Backend.AVRO:
+            return "long"
+        else:
+            return "bigint"
+
+
+class FloatType(DataType):
+    @staticmethod
+    def random_value(record_size: RecordSize) -> Any:
+        if record_size == RecordSize.TINY:
+            return random.random()
+        elif record_size == RecordSize.SMALL:
+            return random.uniform(-100, 100)
+        elif record_size == RecordSize.MEDIUM:
+            return random.uniform(-1_000_000, 1_000_000)
+        elif record_size == RecordSize.LARGE:
+            return random.uniform(-1_000_000_000, 1_000_000_000_00)
+        else:
+            raise ValueError(f"Unexpected record size {record_size}")
+
+    @staticmethod
+    def num_value(num: int) -> Any:
+        return num
+
+    @staticmethod
+    def name(backend: Backend) -> str:
+        if backend == Backend.AVRO:
+            return "float"
+        else:
+            return "float4"
+
+
+class DoubleType(DataType):
+    @staticmethod
+    def random_value(record_size: RecordSize) -> Any:
+        if record_size == RecordSize.TINY:
+            return random.random()
+        elif record_size == RecordSize.SMALL:
+            return random.uniform(-100, 100)
+        elif record_size == RecordSize.MEDIUM:
+            return random.uniform(-1_000_000, 1_000_000)
+        elif record_size == RecordSize.LARGE:
+            return random.uniform(-1_000_000_000, 1_000_000_000_00)
+        else:
+            raise ValueError(f"Unexpected record size {record_size}")
+
+    @staticmethod
+    def num_value(num: int) -> Any:
+        return num
+
+    @staticmethod
+    def name(backend: Backend) -> str:
+        if backend == Backend.AVRO:
+            return "double"
+        else:
+            return "float8"
+
+
+class StringType(DataType):
+    @staticmethod
+    def random_value(record_size: RecordSize) -> Any:
+        if record_size == RecordSize.TINY:
+            return random.choice(("foo", "bar", "baz"))
+        elif record_size == RecordSize.SMALL:
+            return "".join(random.choice(string.printable) for _ in range(3))
+        elif record_size == RecordSize.MEDIUM:
+            return "".join(random.choice(string.printable) for _ in range(10))
+        elif record_size == RecordSize.LARGE:
+            return "".join(random.choice(string.printable) for _ in range(100))
+        else:
+            raise ValueError(f"Unexpected record size {record_size}")
+
+    @staticmethod
+    def num_value(num: int) -> Any:
+        return f"key{num}"
+
+    @staticmethod
+    def name(backend: Backend) -> str:
+        if backend == Backend.AVRO:
+            return "string"
+        else:
+            return "text"
+
+
+# TODO: BYTES, RECORD, ENUM, ARRAY, MAP, FIXED
```

Comment:
I do not think you need to generate all of those types as most of the code in the product is agnostic to the type of the data. Strings and integers would be enough already.